app/models/user.rb

```diff
@@ -37,4 +38,16 @@ def send_notification(change)
     ChangeMailer.page(change: change, user: self).deliver_later
   end
 
+  def email_domain_is_approved
+    user_domain = email.strip.split('@')[-1].downcase
```

Comment:
This will error out if email is `''`. I would add a `rescue ''` at the end of the line.